src/main/java/seedu/address/logic/parser/AddFacebookContactParser.java

```diff
@@ -0,0 +1,72 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import facebook4j.Facebook;
+import facebook4j.FacebookException;
+import facebook4j.FacebookFactory;
+import facebook4j.ResponseList;
+import facebook4j.User;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.logic.commands.AddFacebookContactCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Parses the given {@code String} of arguments in the context of the AddFacebookContactCommand
+ * and returns an AddFacebookContactCommand object for execution.
+ * @throws ParseException if the user input does not conform the expected format
+ */
+public class AddFacebookContactParser implements Parser<AddFacebookContactCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddFacebookContactCommand
+     * and returns an AddFacebookContactCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddFacebookContactCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddFacebookContactCommand.MESSAGE_USAGE));
+        }
+
+        try {
+            Facebook facebook = new FacebookFactory().getInstance();
+            ResponseList<User> contactsList = facebook.searchUsers(trimmedArgs);
+            User user = contactsList.get(0);
+
+            Email email;
+            // create new person object
+            if (user.getEmail() != null) {
+                email = new Email(user.getEmail());
+            } else {
+                email = new Email("a@b.c");
+            }
+
+            Set<Tag> taglist = new HashSet<Tag>();
+            taglist.add(new Tag("facebookcontact"));
+
+            Person newPerson = new Person(new Name(user.getName()), new Phone("000"), email,
```

Comment:
After merging my Favorite Contacts PR, `Person` constructor now requires one new parameter and changes should look like this:

`Person newPerson = new Person(new Name(user.getName()), new Phone("000"), email, new Address("-"), new Favorite(false), taglist);`

Hence, every new Facebook contact person added will not be a favorite contact by default.

---
Travis reported:
```
:compileJava/home/travis/build/CS2103AUG2017-W10-B2/main/src/main/java/seedu/address/logic/parser/AddFacebookContactParser.java:58: error: no suitable constructor found for Person(Name,Phone,Email,Address,Set<Tag>)
            Person newPerson = new Person(new Name(user.getName()), new Phone("000"), email,
                               ^
    constructor Person.Person(Name,Phone,Email,Address,Favorite,Set<Tag>) is not applicable
      (actual and formal argument lists differ in length)
```